src/utils/checkFunction.cpp

```diff
@@ -215,13 +374,120 @@ bool TVM_DLLAPI checkNormalAcceleration(FunctionPtr f, CheckOptions opt)
   }
 }
 
+bool checkNormalAcceleration(const UpdatelessFunction & uf, const CheckOptions & opt)
+{
+  const auto & x = uf.variables();
+
+  int n = 0;
+  int nd = 0;
+  for(const auto & xi : x)
+  {
+    if(!xi->isEuclidean())
+    {
+      // For now, we only accept Euclidean variables. See note in checkJacobian's code.
+      throw std::runtime_error("This function is implemented for Euclidean variables only.");
+    }
+    n += xi->size();
+    nd += dot(xi)->size();
+  }
+
+  // value of x(t)
+  VectorXd val0 = VectorXd::Zero(n);
+  // value of dx(t)
+  VectorXd vel0 = VectorXd::Zero(nd);
+  // value of ddx(t)
+  VectorXd acc = VectorXd::Zero(nd);
+
+  size_t failed = 0;
+  for(size_t i = 0; i < opt.samples; ++i)
+  {
+    val0.setRandom();
+    vel0.setRandom();
+    acc.setRandom().normalize();
+    failed += !checkNormalAcceleration(uf, opt, val0, vel0, acc);
+  }
+  size_t failed_fixed = 0;
+  size_t total_fixed = 0;
+  using CheckConfiguration = CheckOptions::CheckConfiguration;
+  const char * prefix = "[checkNormalAcceleration]";
+  auto getValue = [&](const CheckConfiguration & c) {
+    return getFromConfigOrRandom<&CheckConfiguration::value>(prefix, c, val0);
+  };
+  auto getVelocity = [&](const CheckConfiguration & c) {
+    return getFromConfigOrRandom<&CheckConfiguration::velocity>(prefix, c, vel0);
+  };
+  auto getAcceleration = [&](const CheckConfiguration & c) {
+    return getFromConfigOrRandom<&CheckConfiguration::acceleration>(prefix, c, acc);
+  };
+  auto handleConfiguration = [&](const CheckOptions::CheckConfiguration & c) {
+    if(c.value && c.velocity && c.acceleration)
+    {
+      if(getValue(c) && getVelocity(c) && getAcceleration(c))
+      {
+        total_fixed++;
+        failed_fixed += !checkNormalAcceleration(uf, opt, *c.value, *c.velocity, *c.acceleration);
+      }
+    }
+    else
+    {
+      for(size_t i = 0; i < c.samples; ++i)
+      {
+        if(getValue(c) && getVelocity(c) && getAcceleration(c))
+        {
+          total_fixed++;
+          failed_fixed += !checkNormalAcceleration(uf, opt, val0, vel0, acc);
+        }
+      }
+    }
+  };
+  for(const auto & c : opt.configs)
+  {
+    handleConfiguration(c);
+  }
+  if(opt.verbose && (failed != 0 || failed_fixed != 0))
+  {
+    if(opt.samples != 0 && failed != 0)
+    {
+      std::cout << failed << " random configurations failed out of " << opt.samples << "\n";
+    }
+    if(total_fixed != 0 && failed_fixed != 0)
+    {
+      std::cout << failed_fixed << " configurations failed out of " << total_fixed << "\n";
+    }
+  }
+  return (failed + failed_fixed) == 0;
+}
+
+} // namespace
+
+bool TVM_DLLAPI checkNormalAcceleration(FunctionPtr f, CheckOptions opt)
```

Comment:
The TVM_DLLAPI should not be necessary here.

Reply:
Done in f5f60233